catalog/task/add-bundles-to-image-index/0.1/README.md

```diff
@@ -0,0 +1,21 @@
+# add-bundles-to-image-index
+
+Calls IIB service to submit a build request to add operator bundles to an index image
+
+## Parameters
+
+| Name | Description | Optional | Default value |
+|------|-------------|----------|---------------|
+| IIB_URL | IIB Service URL | No | - |
+| IIB_JSON_BUILD_ARGS | IIB request arguments in JSON format | No | 0 |
+| IIB_SECRET | Secret name that contains IIB credentials and Registry| Yes | 0 |
+
+### Secret parameters
```

Comment:
I am confused as to what a secret parameter is or how it is provided/used. Can you add a short snippet in this section explaining that?

Reply:
It actually refers to what is needed to store in the secret. But sure, I agree a snippet would be useful. Not only for this one, but also for the `{TOKEN}` part.

Reply:
Can you take a look again if it looks better now?

Reply:
It is significantly improved, thanks. I am still a little confused though on how `TOKEN` relates to `Secret` (the section header). Is that entire section really the token, not the secret? Not sure if I am making sense. Or maybe the same `TOKEN` value is required in both the BUILD_ARGS and IIB_SECRET parameter? It could just be I am unfamiliar with iib so I am slow to understand what you mean

Reply:
The `registry_secret` item  (which refers to  IIB's `overwrite_from_index_token` `/buids/add` call )  is used to authenticate to _quay/docker/catalog_ to push (and pull I think) images. The problem is that this value should be plain text in the JSON body and the entrypoint for the FCB build is the Release Strategy. This way you registry user/pass would be unsecured there.

To fix this, I added these credentials to the secret but it still needs to be added to the JSON body. The simpler way I found to make do this in a secure way (so far) was using this `{TOKEN}` placeholder.

If you have any other idea to secure this I am open to listen!

Reply:
so you run the command starting on line 43 and the output is `IIB_SECRET`?

Reply:
No. The IIB_SECRET value should be defined beforehand and then you run the command or write the yaml for the secret, something like this:

```
apiVersion: v1
kind: Secret
metadata:
  name: m9-secret
  namespace: managed
  annotations:
    kubernetes.io/service-account.name: demo-service-account
data:
    krb5_username: base64 of foo
    krb5_password: base64 of bar
    krb5.conf: base64 of the krb5.conf
    registry_name: base64 of foo:bar
```

Reply:
Oh so the Secret is a kubernetes custom resource secret, not like a secret token? Does the task retrieve it from the cluster then? I wonder how that will work on kcp (similarly to how you modified the EC stuff to not have to retrieve a CR from the cluster)

Reply:
The task mounts the Secret (as you can see here https://github.com/hacbs-release/release-bundles/blob/8decaaa206e3a596c63f34f675e94168c741a122/catalog/task/add-bundles-to-image-index/0.1/add-bundles-to-image-index.yaml#L106) and reads all keys (krb5_username, krb5_password, registry_token and krb5.conf) from this mounted path. 

So, with a given Secret I.E `my-secret` - you set it at `iibSecret` at the ReleaseStrategy. This will be translated to the `$(params.IIB_SECRET)` of the task. So this secret will be mounted.

This secret should live in the same workspace/namespace  that the task will run and the service account should have access to it.

Reply:
I'm not familiar enough with how volumes work, I just know that the pipelinerun can, and likely will, be farmed out to a different cluster so there is no guarantee that the namespace/workspace where the pipelinerun is started from is where it runs

Reply:
I think I got what you mean. If the FBC pipelinerun running in a different cluster, how would the secret be accessed by the other cluster. This was not being covered by this ticket and probably the design to make it work will be totally different.

Reply:
Right. I asked on slack to see how this works with volume mounts. @scoheb can correct me here but I am not sure we are planning to run any future milestone demos on standalone clusters, only kcp

Reply:
Secrets should work with KCP as they are native stuff. My concern is about the FBC implementation of running the full Pipeline in another Cluster. KCP workspaces/namespaces are not a problem.

Reply:
It sounds like this pipeline will not run on kcp (sorry for all the confusion) so I am going to resolve this